shared/src/Datadog.Trace.ClrProfiler.Native/dllmain.cpp

```diff
@@ -14,67 +26,90 @@ using namespace datadog::shared::nativeloader;
+
+            isLogDebugEnabled = IsLogDebugEnabledDefault;
+        }
+        else
+        {
+            if (!shared::TryParseBooleanEnvironmentValue(isLogDebugEnabledStr, isLogDebugEnabled))
             {
-                Log::Info("No \"", EnvironmentVariables::DebugLogEnabled, "\" environment variable has been found.",
-                          " Enable debug log = ", IsLogDebugEnabledDefault, " (default).");
+                // invalid value for environment variable
+                Log::Info("Non boolean value \"", isLogDebugEnabledStr, "\" for \"",
+                    EnvironmentVariables::DebugLogEnabled, "\" environment variable.",
+                    " Enable debug log = ", IsLogDebugEnabledDefault, " (default).");
 
                 isLogDebugEnabled = IsLogDebugEnabledDefault;
             }
             else
             {
-                if (!shared::TryParseBooleanEnvironmentValue(isLogDebugEnabledStr, isLogDebugEnabled))
-                {
-                    // invalid value for environment variable
-                    Log::Info("Non boolean value \"", isLogDebugEnabledStr, "\" for \"",
-                              EnvironmentVariables::DebugLogEnabled, "\" environment variable.",
-                              " Enable debug log = ", IsLogDebugEnabledDefault, " (default).");
-
-                    isLogDebugEnabled = IsLogDebugEnabledDefault;
-                }
-                else
-                {
-                    // take environment variable into account
-                    Log::Info("Enable debug log = ", isLogDebugEnabled, " from (", EnvironmentVariables::DebugLogEnabled, " environment variable)");
-                }
+                // take environment variable into account
+                Log::Info("Enable debug log = ", isLogDebugEnabled, " from (", EnvironmentVariables::DebugLogEnabled, " environment variable)");
             }
+        }
 
-            if (isLogDebugEnabled)
-            {
-                Log::EnableDebug(true);
-            }
+        if (isLogDebugEnabled)
+        {
+            Log::EnableDebug(true);
+        }
 
-            Log::Debug("DllMain: DLL_PROCESS_ATTACH");
-            Log::Debug("DllMain: Pointer size: ", 8 * sizeof(void*), " bits.");
+#if CRASHTRACKING
+        bool telemetry_enabled = true;
+        shared::TryParseBooleanEnvironmentValue(shared::GetEnvironmentValue(L"DD_INSTRUMENTATION_TELEMETRY_ENABLED"), telemetry_enabled);
 
-            dispatcher = new DynamicDispatcherImpl();
-            dispatcher->LoadConfiguration(GetConfigurationFilePath());
+        bool crashtracking_enabled = true;
+        shared::TryParseBooleanEnvironmentValue(shared::GetEnvironmentValue(L"DD_CRASHTRACKING_ENABLED"), crashtracking_enabled);
 
-            // *****************************************************************************************************************
-            break;
+        if (telemetry_enabled && crashtracking_enabled)
```

Comment:
why is it needed that the telemetry is enabled to send the crash report?

Reply:
By design. We consider the crash report to be telemetry.